compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2405,6 +2438,34 @@ fn add_upstream_rust_crates<'a>(
                 // external build system already has the native dependencies defined, and it
                 // will provide them to the linker itself.
                 if sess.opts.unstable_opts.link_native_libraries {
+                    if sess.opts.unstable_opts.separate_native_rlib_dependencies {
+                        // if rlib contains native libs as archieves, unpack them to tmpdir
+                        let archive_map = unsafe {
+                            Mmap::map(File::open(&src.rlib.as_ref().unwrap().0).unwrap()).unwrap()
```

Comment:
Operations that can realistically fail like `File::open` or `Mmap::map` should use `sess.fatal(...)` for errors instead of unwrapping.
See e.g. `fn load_metadata_with` for an example.

Reply:
Same applies to other unwraps below.